scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/mirrors/OfflineMirror.scala

```diff
@@ -0,0 +1,56 @@
+package scala.meta
+package internal
+package semantic
+package mirrors
+
+import java.io._
+import org.scalameta.data._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.meta.internal.io.PlatformIO
+import scala.meta.parsers._
+import scala.meta.semantic._
+
+@data class OfflineMirror(classpath: String, sourcepath: String) extends CommonMirror {
```

Comment:
Following discussions in #777 

I propose that this definition becomes
```scala
@data class OfflineMirror(database: Database, sources: Seq[Source], dialect: Dialect = Scala212) extends CommonMirror // no body implementation

object OfflineMirror {
  def apply(classpath: String, sourcepath: String): Try[OfflineMirror]
}
```
This way an OfflineMirror becomes a true `@data` class. 

In sbt-scalafix, I will probably want to construct an OfflineMirror from a single `.semanticdb` file + `sourceRoot: AbsolutePath`. The current `classpath: String, sourcepath: String` primary constructor is not suitable for that use-case.

Note. My current workaround is to do 
```scala
new CommonMirror {
  override dialect ...
  override sources
  override database
}
```